tests/integration/targets/aci_cloud_region/tasks/main.yml

```diff
@@ -10,57 +10,67 @@
 
 # CLEAN ENVIRONMENT
 - name: Set vars
-  set_fact: 
+  set_fact:
     aci_info: &aci_info
-      host: '{{ aws_aci_hostname }}'
-      username: '{{ aws_aci_username }}'
-      password: '{{ aws_aci_password }}'
+      host: "{{aci_hostname}}"
+      username: "{{aci_username}}"
+      password: "{{aci_password}}"
       validate_certs: '{{ aci_validate_certs | default(false) }}'
       use_ssl: '{{ aci_use_ssl | default(true) }}'
       use_proxy: '{{ aci_use_proxy | default(true) }}'
       output_level: '{{ aci_output_level | default("info") }}'
 
-- name: Query all regions
-  aci_cloud_region:
-    <<: *aci_info
-    cloud: 'aws'
-    state: query
-  register: query_all
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
 
-- name: Verify query_all
-  assert:
-    that:
-    - query_all is not changed
-    - query_all.current.0.cloudProvP.attributes.dn == "uni/clouddomp/provp-aws"
-    - query_all.current.0.cloudProvP.attributes.vendor == "aws"
-    - query_all.current.0.cloudProvP.children | length > 10
+- name: Execute tasks only for cloud sites
+  when:
+  - version.current.0.topSystem.attributes.version is version('4.1', '>=')
+  - query_cloud.current | length > 0   # This condition will skip execution for non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Query all regions
+    aci_cloud_region:
+      <<: *aci_info
+      cloud: "{{cloud_type}}"
+      state: query
+    register: query_all
 
-- name: Query a specific region
-  aci_cloud_region:
-    <<: *aci_info
-    cloud: 'aws'
-    region: eu-west-2
-    state: query
-  register: query_region
+  - name: Verify query_all
+    assert:
+      that:
+      - query_all is not changed
+      - query_all.current.0.cloudProvP.attributes.dn == "uni/clouddomp/provp-{{cloud_type}}"
+      - query_all.current.0.cloudProvP.attributes.vendor == "{{cloud_type}}"
+      - query_all.current.0.cloudProvP.children | length > 10
+    when: query_all.current != []
 
-- name: Verify query_region
-  assert:
-    that:
-    - query_region is not changed
-    - query_region.current.0.cloudRegion.attributes.adminSt == "unmanaged"
-    - query_region.current.0.cloudRegion.attributes.dn == "uni/clouddomp/provp-aws/region-eu-west-2"
-    - query_region.current.0.cloudRegion.attributes.name == "eu-west-2"
+  - name: Query a specific region
+    aci_cloud_region:
+      <<: *aci_info
+      cloud: "{{cloud_type}}"
+      region: eu-west-2
```

Comment:
why not use one of the region you already have a variable for?
That would make it work for both Azure and AWS no?

Reply:
Made region generic